test/storage/index.test.ts

```diff
@@ -0,0 +1,515 @@
+        const { client, headers } = clientAndHeaders();
+
+        // Act
+        const body = await client.moveFolder(payload);
+
+        // Assert
+        expect(callStub.firstCall.args).toEqual([
+          '/api/storage/move/folder',
+          {
+            folderId: payload.folder.id,
+            destination: payload.destinationFolderId,
+          },
+          {
+            headers: headers
+          }
+        ]);
+        expect(body).toEqual(moveFolderResponse);
+      });
+
+    });
+
+    describe('update folder metadata', () => {
+
+      it('Should bubble up the error', async () => {
+        // Arrange
+        const payload = randomUpdateFolderMetadataPayload();
+        const { client } = clientAndHeaders();
+        sinon.stub(axios, 'post').rejects(new Error('first call error'));
+
+        // Act
+        const call = client.updateFolder(payload);
+
+        // Assert
+        await expect(call).rejects.toThrowError('first call error');
+      });
+
+      it('Should call with right params & return correct response', async () => {
+        // Arrange
+        const payload = randomUpdateFolderMetadataPayload();
+        const callStub = sinon.stub(axios, 'post').resolves(validResponse({}));
+        const { client, headers } = clientAndHeaders();
+
+        // Act
+        await client.updateFolder(payload);
+
+        // Assert
+        expect(callStub.firstCall.args).toEqual([
+          '/api/storage/folder/2/meta',
+          {
+            metadata: {
+              itemName: payload.changes.itemName,
+              color: payload.changes.color,
+              icon: payload.changes.icon,
+            }
+          },
+          {
+            headers: headers
+          }
+        ]);
+      });
+
+    });
+
+    describe('delete folder', () => {
+
+      it('Should bubble up the error', async () => {
+        // Arrange
+        sinon.stub(axios, 'delete').rejects(new Error('first call error'));
+        const { client } = clientAndHeaders();
+
+        // Act
+        const call = client.deleteFolder(2);
+
+        // Assert
+        await expect(call).rejects.toThrowError('first call error');
+      });
+
+      it('Should call with right arguments & return content', async () => {
+        // Arrange
+        const callStub = sinon.stub(axios, 'delete').resolves(validResponse({
+          valid: true
+        }));
+        const { client, headers } = clientAndHeaders();
+
+        // Act
+        const body = await client.deleteFolder(2);
+
+        // Assert
+        expect(callStub.firstCall.args).toEqual([
+          '/api/storage/folder/2',
+          {
+            headers: headers
+          }
+        ]);
+        expect(body).toEqual({
+          valid: true
+        });
+      });
+
+    });
+
+  });
+
+  describe('-> files', () => {
+
+    describe('create file entry', () => {
+
+      it('Should have all the correct params on call', async () => {
```

Comment:
Add a test to enforce encrypt_version === "03-aes"

Reply:
I didn't add the text explicitly, but since the payload expects an EncryptionVersion type and it is defined as the only possible type, the checking is implicit.